megalista_dataflow/uploaders/google_ads/customer_match/abstract_uploader.py

```diff
@@ -14,206 +14,212 @@
-
-  def _do_create_list_if_it_does_not_exist(self,
-                                           customer_id: str,
-                                           list_name: str,
-                                           list_definition: Dict[str, Any]
-                                           ) -> str:
-    
-    resource_name = self._get_user_list_resource_name(customer_id, list_name)
-    
-    if resource_name is None:
-      # Create list
-      logging.getLogger(_DEFAULT_LOGGER).info(
-        '%s list does not exist, creating...', list_name)
-      request = {
-        'customer_id': customer_id,
-        'partial_failure': False,
-        'validate_only': False,
-        'operations': [{
-          'create': list_definition
-        }]
-      }
-      
-      user_list_service = self._get_user_list_service(customer_id)
-      user_list_service_response = user_list_service.mutate_user_lists(request)
-      for result in user_list_service_response.results:
-        resource_name = result.resource_name
-      logging.getLogger(_DEFAULT_LOGGER).info('List %s created with resource name: %s',
-                                              list_name, resource_name)
-    else:
-      logging.getLogger(_DEFAULT_LOGGER).info('List %s found with resource name: %s',
-                                              list_name, resource_name)
-    return str(resource_name)
-
-  def _get_user_list_resource_name(self, customer_id: str, list_name: str) -> Optional[str]:
-    ads_client = utils.get_ads_client(self.oauth_credentials, self.developer_token.get(), customer_id)
-
-    resource_name = None
-    service = self._get_ads_service(customer_id)
-
-    # Only search for audiences owned by this account, not MCCs above it.
-    query = f"SELECT user_list.resource_name, user_list.access_reason FROM user_list WHERE user_list.name='{list_name}' " \
-            f"AND user_list.access_reason={ads_client.enums.AccessReasonEnum.OWNED.name}"
-    response_query = service.search_stream(customer_id=customer_id, query=query)
-    for batch in response_query:
-      for row in batch.results:
-        resource_name = row.user_list.resource_name
-    return resource_name
-
-  # just to facilitate mocking
-  def _get_ads_service(self, customer_id: str):
-    return utils.get_ads_service('GoogleAdsService', ADS_API_VERSION,
+    def __init__(self, oauth_credentials: OAuthCredentials, developer_token: StaticValueProvider):
+        super().__init__()
+        self.oauth_credentials = oauth_credentials
+        self.developer_token = developer_token
+        self.active = True
+        if self.developer_token is None:
+            self.active = False
+        self._user_list_id_cache: Dict[str, str] = {}
+        self._job_cache: Dict[str, Dict[str, str]] = {}
+
+    def start_bundle(self):
+        pass
+
+    def finish_bundle(self):
+        for job_definition in self._job_cache.values():
+            print(f"Running job {job_definition['job_resource_name']}")
```

Comment:
Replace for logging if you can, I think "print" doesn't show up in Dataflow logs.